### PR TITLE
[feat] swagger for auth routes

### PR DESCRIPTION
**Pull request by Tristancalard**

### CODE REVIEW

**Review by thmarinho — CHANGES_REQUESTED** [review]
image

File: backend/src/app.ts [hunk 1]
```diff
@@ -7,6 +7,24 @@ import dotenv from 'dotenv';
 import usersRouter from './routes/users/router_users.js';
 import authRouter from './routes/auth/auth.js';
 
+import swaggerJSDoc from 'swagger-jsdoc';
+import swaggerUi from 'swagger-ui-express';
+
+const swaggerDefinition = {
+  openapi: '3.0.0',
+  info: {
+    title: 'Express API',
```
Comment (thmarinho) [inline backend/src/app.ts]: Use a proper title

File: backend/src/app.ts [hunk 2]
```diff
@@ -18,8 +36,8 @@ app.use(
     resave: false,
     saveUninitialized: false,
     cookie: {
-      httpOnly: true,
-      secure: false,
+    httpOnly: true,
+    secure: false,
```
Comment (thmarinho) [inline backend/src/app.ts]: Fix identation

File: backend/src/app.ts [hunk 3]
```diff
@@ -31,6 +49,7 @@ app.use(express.urlencoded({ extended: false }));
 
 app.use('/users', usersRouter);
 app.use('/auth', authRouter);
+app.use('/docs', swaggerUi.serve, swaggerUi.setup(swaggerSpec, { explorer: true }));
```
Comment (thmarinho) [inline backend/src/app.ts]: use the /swagger route instead

File: backend/src/app.ts [hunk 4]
```diff
@@ -7,6 +7,24 @@ import dotenv from 'dotenv';
...
+    version: '1.0.0',
+  },
+};
+
+const options = {
+  swaggerDefinition,
+  apis: [`dist/routes/**/*.js`],
+};
+
+const swaggerSpec = swaggerJSDoc(options);
```
Comment (thmarinho) [inline backend/src/app.ts]: Don't declare options in a variable, just send the object in the `swaggerJSDoc`

**Review by thmarinho — CHANGES_REQUESTED** [review]

File: backend/src/app.ts [hunk 5]
```diff
@@ -7,6 +7,22 @@ import dotenv from 'dotenv';
 import usersRouter from './routes/users/router_users.js';
 import authRouter from './routes/auth/auth.js';
 
+import swaggerJSDoc from 'swagger-jsdoc';
+import swaggerUi from 'swagger-ui-express';
```
Comment (thmarinho) [inline backend/src/app.ts:7]: Move this with the other package imports

File: backend/src/app.ts [hunk 6]
```diff
@@ -7,6 +7,22 @@ import dotenv from 'dotenv';
...
+    title: 'Plateforme HUB',
+    version: '1.0.0',
+  },
+};
+
+const swaggerSpec = swaggerJSDoc({
+  swaggerDefinition,
+  apis: [`dist/routes/**/*.js`],
+});
+
```
Comment (thmarinho) [inline backend/src/app.ts]: Don't declare `swaggerDefinition` as a variable, just send the object in the `swaggerJSDoc`

File: .gitignore [hunk 7]
```diff
@@ -74,7 +74,6 @@ web_modules/
 
 # dotenv environment variable files
 .env
-.env.dev
```
Comment (thmarinho) [inline .gitignore]: Keep this line

File: backend/.env.template [hunk 8]
```diff
@@ -5,4 +5,13 @@ PGPASSWORD=
 ENDPOINT_ID=
 PGPORT=
 
-NODE_ENV=
\ No newline at end of file
+NODE_ENV=
+
```
Comment (thmarinho) [inline backend/.env.template]: Remove empty lines

File: backend/src/routes/auth/auth.ts [hunk 9]
```diff
@@ -11,20 +11,52 @@ import { REDIRECT_URI, POST_LOGOUT_REDIRECT_URI } from '../../config/authConfig.
...
+ *   get:
+ *     summary: Microsoft connection.
+ *     description: Allows you to connect using your Microsoft account and redirects to /auth/token to retrieve the token.
+*/
+
 router.get('/signin', authProvider.login({
     scopes: ['User.Read'],
     redirectUri: REDIRECT_URI,
-    successRedirect: '/'
+    successRedirect: '/auth/token'
```
Comment (thmarinho) [inline backend/src/routes/auth/auth.ts:25]: Did you try this configuration ? Changing a redirectUrl can't broke a lot of things

File: backend/src/routes/auth/auth.ts [hunk 10]
```diff
@@ -11,20 +11,52 @@ import { REDIRECT_URI, POST_LOGOUT_REDIRECT_URI } from '../../config/authConfig.
...
+ *   get:
+ *     summary: Retrieve the token.
+ *     description: Retrieve the token and becomes accessible from req.session.accessToken.
+*/
+
 router.get('/token', authProvider.acquireToken({
     scopes: ['User.Read'],
     redirectUri: REDIRECT_URI,
-    successRedirect: '/users/profile'
+    successRedirect: '/'
```
Comment (thmarinho) [inline backend/src/routes/auth/auth.ts:39]: Same question here

**Review by thmarinho — CHANGES_REQUESTED** [review]

File: backend/src/routes/auth/auth.ts (hunk 9 above)
Comment (Tristancalard) [reply backend/src/routes/auth/auth.ts:25]: yes it's work

File: backend/src/routes/auth/auth.ts (hunk 10 above)
Comment (Tristancalard) [reply backend/src/routes/auth/auth.ts:39]: yes it's work

File: backend/src/app.ts (hunk 6 above)
Comment (Tristancalard) [reply backend/src/app.ts]: Error: Missing or invalid input: 'options.swaggerDefinition' or 'options.definition' is required
 I need it

Comment (thmarinho) [reply backend/src/app.ts]:
```js
const swaggerSpec = swaggerJSDoc({
  swaggerDefinition : {
  openapi: '3.0.0',
  info: {
    title: 'Plateforme HUB',
    version: '1.0.0',
  },
},
  apis: [`dist/routes/**/*.js`],
});
```

**Review by thmarinho — CHANGES_REQUESTED** [review]

**Review by thmarinho — APPROVED** [review]

**Review by alanschnegg — APPROVED** [review]

**Pull request merged**